src/Fhir.Anonymizer.Core/Resource/ResourceIdTransformer.cs

```diff
@@ -0,0 +1,134 @@
+            new Regex(@"urn:oid:(?<id>[0-2](\.(0|[1-9][0-9]*))+)"),
+            new Regex(@"urn:uuid:(?<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")
+        };
+
+        private static readonly ConcurrentDictionary<string, string> _resourceIdMap = new ConcurrentDictionary<string, string>();
+        private static readonly ILogger _logger = AnonymizerLogging.CreateLogger<AnonymizerEngine>();
+
+        public static void Transform(ElementNode node)
+        {
+            if (ModelInfo.IsKnownResource(node.InstanceType))
+            {
+                var idNode = node.Children("id").Cast<ElementNode>().FirstOrDefault();
+                if (idNode != null)
+                {
+                    var newId = TransformId(idNode.Value?.ToString());
+                    _logger.LogDebug($"Resource Id {idNode.Value?.ToString()} is transformed to {newId}");
+                    idNode.Value = newId;
+                }
+            }
+            else if (ModelInfo.IsReference(node.InstanceType))
+            {
+                var referenceNode = node.Children("reference").Cast<ElementNode>().FirstOrDefault();
+                if (referenceNode != null)
+                {
+                    var newReference = TransformIdFromReference(referenceNode.Value?.ToString());
+                    referenceNode.Value = newReference;
+                }
+            }
+
+            foreach(var child in node.Children().Cast<ElementNode>())
+            {
+                Transform(child);
+            }
+        }
+
+        public static void SaveMappingFile(string mappingFile)
+        {
+            using var fileStream = new FileStream(mappingFile, FileMode.Create);
+            using var writer = new StreamWriter(fileStream);
+            foreach(var k in _resourceIdMap.Keys)
+            {
+                writer.WriteLine($"{k}{MappingFileDelimiter}{_resourceIdMap[k]}");
```

Comment:
Can we just write json string of dictionary?

Reply:
Then we can avoid parse it ourself.

Reply:
Changed to json serialization/deserialization, thanks